settings/generic_setting.go

```diff
@@ -199,6 +199,100 @@ func (w accountSetting[T]) GetId(t *T) string {
 
 var _ accountSettingDefinition[struct{}] = accountSetting[struct{}]{}
 
+type accountWorkspaceSettingDefinition[T any] genericSettingDefinition[T, *common.DatabricksClient]
+
+// An account workspace setting is a setting that can be scoped to either an account or a workspace.
+type accountWorkspaceSetting[T any] struct {
+	// The struct corresponding to the setting. The schema of the Terraform resource will be generated from this struct.
+	// This struct must have an Etag field of type string.
+	settingStruct T
+
+	// Read the setting from the server. The etag is provided as the third argument.
+	readAccFunc func(ctx context.Context, acc *databricks.AccountClient, etag string) (*T, error)
+
+	readWsFunc func(ctx context.Context, w *databricks.WorkspaceClient, etag string) (*T, error)
+
+	// Update the setting to the value specified by t, and return the new etag. If the setting name is user-settable,
+	// it will be provided in the third argument. If not, you must set the SettingName field appropriately. You must
+	// also set AllowMissing: true and the field mask to the field to update.
+	updateAccFunc func(ctx context.Context, w *databricks.AccountClient, setting T) (string, error)
+
+	updateWsFunc func(ctx context.Context, w *databricks.WorkspaceClient, setting T) (string, error)
+
+	// Delete the setting with the given etag, and return the new etag.
+	deleteAccFunc func(ctx context.Context, acc *databricks.AccountClient, etag string) (string, error)
+
+	deleteWsFunc func(ctx context.Context, w *databricks.WorkspaceClient, etag string) (string, error)
+
+	// Optional function to generate resource ID from the settings. If not provided, will use predefined value `global`
+	generateIdFunc func(setting *T) string
+}
+
+func (aw accountWorkspaceSetting[T]) SettingStruct() T {
+	return aw.settingStruct
+}
+func (aw accountWorkspaceSetting[T]) Read(ctx context.Context, c *common.DatabricksClient, etag string) (*T, error) {
+	if c.Config.IsAccountClient() {
```

Comment:
We can use `c.AccountOrWorksapceRequest` for all of these.

Reply:
`AccountOrWorkspaceRequest` is a method on `DatabricksClient`, so won't accept a type parameter 😢